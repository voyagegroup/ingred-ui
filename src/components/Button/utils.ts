import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
