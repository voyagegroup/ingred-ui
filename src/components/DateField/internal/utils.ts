export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
