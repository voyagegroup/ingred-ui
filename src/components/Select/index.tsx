export { default } from "./Select";
export type { SelectProps, OptionType } from "./Select";
