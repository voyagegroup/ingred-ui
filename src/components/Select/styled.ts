import styled from "styled-components";
import { components, MenuListProps } from "react-select";
import { addScrollbarProperties } from "../../utils/scrollbar";
import { OptionType } from "./Select";

export const Container = styled.div<{
  minWidth?: string;
  isDisabled?: boolean;
}>`
  min-width: ${({ minWidth }) => minWidth || "auto"};
  cursor: ${({ isDisabled }) => (isDisabled ? "not-allowed" : "auto")};
`;

export const ReactSelectMenuList = styled(components.MenuList)<
  MenuListProps<OptionType<any>, boolean>
>`
  ${({ maxHeight }) => addScrollbarProperties({ maxHeight: `${maxHeight}px` })}
` as any; // TODO: remove any
