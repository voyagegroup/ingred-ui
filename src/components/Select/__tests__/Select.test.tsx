import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import Select from "../";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import { OptionType } from "../Select";

describe("Select component testing", () => {
  afterEach(cleanup);

  const options: OptionType[] = [
    { label: "hoge", value: "1" },
    { label: "huga", value: "2" },
  ];

  describe("Normal", () => {
    test("not selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select menuIsOpen options={options} />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test("one selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select
          menuIsOpen
          options={options}
          defaultValue={{ label: "hoge", value: "1" }}
        />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test("multiple selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select
          menuIsOpen
          isMulti
          options={options}
          defaultValue={[
            { label: "hoge", value: "1" },
            { label: "huga", value: "2" },
          ]}
        />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("Disable", () => {
    test("not selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select menuIsOpen isDisabled options={options} />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test("one selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select
          menuIsOpen
          isDisabled
          options={options}
          defaultValue={{ label: "hoge", value: "1" }}
        />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test("multiple selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select
          menuIsOpen
          isDisabled
          isMulti
          options={options}
          defaultValue={[
            { label: "hoge", value: "1" },
            { label: "huga", value: "2" },
          ]}
        />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("Error", () => {
    test("not selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select menuIsOpen error options={options} />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test("one selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select
          menuIsOpen
          error
          options={options}
          defaultValue={{ label: "hoge", value: "1" }}
        />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
    test("multiple selected", () => {
      const { asFragment } = renderWithThemeProvider(
        <Select
          menuIsOpen
          error
          isMulti
          options={options}
          defaultValue={[
            { label: "hoge", value: "1" },
            { label: "huga", value: "2" },
          ]}
        />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
