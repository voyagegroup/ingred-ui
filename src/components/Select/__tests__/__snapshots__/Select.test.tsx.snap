// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component testing Disable multiple selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-14jk2my-container"
    >
      <div
        class=" css-7n5mtx-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class="css-jahdpg-multiValue"
          >
            <div
              class="css-uybk9d"
            >
              hoge
            </div>
            <div
              class="css-19h014q"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-jahdpg-multiValue"
          >
            <div
              class="css-uybk9d"
            >
              huga
            </div>
            <div
              class="css-19h014q"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                disabled=""
                id="react-select-7-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#B3BAC1"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-4b0kah-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-10hvqfk-NoOptionsMessage"
          >
            見つかりませんでした
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Disable not selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-14jk2my-container"
    >
      <div
        class=" css-7n5mtx-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class=" css-dcdqlp-placeholder"
          >
            選択してください
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                disabled=""
                id="react-select-5-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#B3BAC1"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-4b0kah-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-64ovdt-option"
            id="react-select-5-option-0"
            tabindex="-1"
          >
            hoge
          </div>
          <div
            class=" css-64ovdt-option"
            id="react-select-5-option-1"
            tabindex="-1"
          >
            huga
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Disable one selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-14jk2my-container"
    >
      <div
        class=" css-7n5mtx-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class=" css-10spk3r-singleValue"
          >
            hoge
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                disabled=""
                id="react-select-6-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#B3BAC1"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-4b0kah-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-1sv3iew-option"
            id="react-select-6-option-0"
            tabindex="-1"
          >
            hoge
          </div>
          <div
            class=" css-64ovdt-option"
            id="react-select-6-option-1"
            tabindex="-1"
          >
            huga
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Error multiple selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-15ys67v-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class="css-jahdpg-multiValue"
          >
            <div
              class="css-18bjtpy"
            >
              hoge
            </div>
            <div
              class="css-19h014q"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-jahdpg-multiValue"
          >
            <div
              class="css-18bjtpy"
            >
              huga
            </div>
            <div
              class="css-19h014q"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-10-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            class="sc-eCssSg gIOVBe"
          >
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#EB0A4E"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-1j004f1-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-10hvqfk-NoOptionsMessage"
          >
            見つかりませんでした
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Error not selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-15ys67v-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class=" css-2rfliy-placeholder"
          >
            選択してください
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-8-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#EB0A4E"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-1j004f1-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-64ovdt-option"
            id="react-select-8-option-0"
            tabindex="-1"
          >
            hoge
          </div>
          <div
            class=" css-64ovdt-option"
            id="react-select-8-option-1"
            tabindex="-1"
          >
            huga
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Error one selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-15ys67v-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class=" css-etdf8s-singleValue"
          >
            hoge
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-9-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            class="sc-eCssSg gIOVBe"
          >
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#EB0A4E"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-1j004f1-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-1sv3iew-option"
            id="react-select-9-option-0"
            tabindex="-1"
          >
            hoge
          </div>
          <div
            class=" css-64ovdt-option"
            id="react-select-9-option-1"
            tabindex="-1"
          >
            huga
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Normal multiple selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-1ukuozi-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class="css-jahdpg-multiValue"
          >
            <div
              class="css-gqt44k"
            >
              hoge
            </div>
            <div
              class="css-19h014q"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-jahdpg-multiValue"
          >
            <div
              class="css-gqt44k"
            >
              huga
            </div>
            <div
              class="css-19h014q"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-4-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            class="sc-eCssSg gIOVBe"
          >
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#041C33"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-4b0kah-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-10hvqfk-NoOptionsMessage"
          >
            見つかりませんでした
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Normal not selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-1ukuozi-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class=" css-b1k5wd-placeholder"
          >
            選択してください
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-2-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#041C33"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-4b0kah-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-64ovdt-option"
            id="react-select-2-option-0"
            tabindex="-1"
          >
            hoge
          </div>
          <div
            class=" css-64ovdt-option"
            id="react-select-2-option-1"
            tabindex="-1"
          >
            huga
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select component testing Normal one selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fQdfBQ"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-1ukuozi-control"
      >
        <div
          class=" css-12ms9px-ValueContainer"
        >
          <div
            class=" css-105lwrm-singleValue"
          >
            hoge
          </div>
          <div
            class="css-1knq81y-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-3-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <div
            class="sc-eCssSg gIOVBe"
          >
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <div
                class="sc-dlfnbm hnsUFa"
                size="12"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <div
              class="sc-hKgILt ivLANu"
            >
              <div
                class="sc-dlfnbm cDOLxW"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0,0H24V24H0Z"
                    fill="none"
                  />
                  <path
                    d="M12,15,7.757,10.757,9.172,9.343,12,12.172l2.828-2.829,1.415,1.414Z"
                    fill="#041C33"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-4b0kah-menu"
      >
        <div
          class=" css-47qful-MenuList"
        >
          <div
            class=" css-1sv3iew-option"
            id="react-select-3-option-0"
            tabindex="-1"
          >
            hoge
          </div>
          <div
            class=" css-64ovdt-option"
            id="react-select-3-option-1"
            tabindex="-1"
          >
            huga
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
